packages/components/src/toggle-group-control/toggle-group-control/component.tsx

```diff
@@ -81,15 +98,29 @@ function UnconnectedToggleGroupControl(
 			) }
 			<MainControl
 				{ ...otherProps }
-				className={ classes }
+				className={ clsx(
+					animationEnabled && 'is-animation-enabled',
+					classes
+				) }
+				style={ {
+					'--indicator-left': indicatorPosition.left,
+					'--indicator-width': indicatorPosition.width,
+					'--indicator-height': indicatorPosition.height,
+					...otherProps.style,
+				} }
+				onTransitionEnd={ ( event ) => {
+					if ( event.pseudoElement === '::before' ) {
```

Comment:
Maybe better to add a code comment explaining that we're using the `pseudoElement` property to make sure we're targeting the right transitionend event?

Reply:
Hmm, I kinda feel like it's somewhat self-explanatory by the context, especially if you take a quick look at the styles. I'm not opposed to adding a comment but it does seem like it'd be a bit redundant to me.